### PR TITLE
Update demos to use Permissions-Policy header syntax

### PR DESCRIPTION
**Pull request by huchenlei**
This PR renames all remaining 'feature policy' occurence with 'permissions policy' in 
- code comments
- function names, variable names
- html content
- README

Header syntax in policies.json has been updated.

Both the 'Permissions-Policy' HTTP header and 'Feature-Policy' HTTP header will be send for backward compatibility.
(See modification in server.mjs)

### CODE REVIEW

File: permissions-policy-header.mjs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    function mapAllowlistItem(item) {
+      switch (item) {
+        case '*':
+          return '*';
+        case "self":
+          return "'self'";
+        case "":
+          return "'none'";
+        default:
+          // Remove leading and tailing '"' in permissions header syntax.
```
Comment (jpchase) [inline permissions-policy-header.mjs:114]: Nit: I think this should be "trailing"
```suggestion
          // Remove leading and trailing '"' in permissions header syntax.
```

File: permissions-policy-header.mjs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      .reduce((acc, item) => {
+        const [policyName, ...allowlist] = item.trim().split(' ').filter(s => s !== '');
+        acc.set(policyName, allowlist);
+        return acc;
+      }, new Map()));
+  }
+
+  serialize() {
+    return [...this.policies.entries()]
+      .map(([policyName, allowlist]) => `${policyName} ${allowlist.join(' ')}`)
```
Comment (jpchase) [inline permissions-policy-header.mjs:46]: Do you need to do the object spreading here (i.e. the `...this.policies.entries()`)? Given that the array is being mapped to create a new array anyways, could the spreading be removed?

```suggestion
return this.policies.entries().map(.....)
```

File: permissions-policy-header.mjs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        case "":
+          return "'none'";
+        default:
+          // Remove leading and tailing '"' in permissions header syntax.
+          return item.slice(1, -1);
+      }
+    }
+    return new FeaturePolicyHeader(
+      new Map(
+        [...this.policies.entries()]
```
Comment (jpchase) [inline permissions-policy-header.mjs:120]: Ditto for not using object spreading.

File: permissions-policy-header.mjs [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+          match_result.length > 1 ? match_result[1].split(' ') : ['']
+        ) : [allowlist_string];
+        acc.set(policyName, allowlist);
+        return acc;
+      }, new Map()));
+  }
+
+  serialize() {
+    return [...this.policies.entries()]
+      .map(([policyName, allowlist]) => `${policyName}=(${allowlist.join(' ')})`)
```
Comment (jpchase) [inline permissions-policy-header.mjs:91]: Ditto for not using object spreading.

File: public/js/app.js [hunk 5]
```diff
@@ -73,7 +73,7 @@ function updatePolicyValue(anchor) {
 /**
  * Updates the dynamic portions of the page.
  * @param {!HTMLAnchorElement} anchor
- * @param {string} id Feature policy id.
+ * @param {string} id Permissions policy id.
```
Comment (jpchase) [inline public/js/app.js:76]: Nit: ditto for applying to both Permissions and Document policy?

File: public/js/shared.js [hunk 6]
```diff
@@ -120,7 +127,7 @@ function policyValueSelector(policy) {
 }
 
 /**
- * Updates the UI metadata header when a feature policy is selected.
+ * Updates the UI metadata header when a permissions policy is selected.
```
Comment (jpchase) [inline public/js/shared.js:130]: Nit: ditto for applying to both Permissions and Document policy?

File: public/js/shared.js [hunk 7]
```diff
@@ -21,9 +21,16 @@ import {ifDefined} from '/lit-html/directives/if-defined.js';
 export const policyOn = new URL(location).searchParams.has('on');
 export const currentPolicyId = new URL(location).pathname.split('/').slice(-1)[0].split('.')[0];
 
-export const featurePolicyAPISupported = 'policy' in document || 'featurePolicy' in document;
+export const permissionsPolicyAPISupported =
+  'policy' in document ||
```
Comment (jpchase) [inline public/js/shared.js:25]: Ditto for not supporting `document.policy`.

File: public/js/shared.js [hunk 8]
```diff
@@ -21,9 +21,16 @@ import {ifDefined} from '/lit-html/directives/if-defined.js';
...
+  'featurePolicy' in document ||
+  'permissionsPolicy' in document;
+
+const permissionsPolicy =
+  document.policy ||
```
Comment (jpchase) [inline public/js/shared.js:30]: Ditto for not supporting `document.policy`.

File: public/js/app.js [hunk 9]
```diff
@@ -102,8 +102,9 @@ function toggleDrawer(forClose = false) {
  * @param {Array<Object>} implementedPolicies
  */
 async function leftOverPolicies(implementedPolicies) {
-  const featurePolicy = document.policy || document.featurePolicy;
-  const allPolicies = featurePolicy.allowedFeatures().sort();
+  const permissionsPolicy =
+    document.policy || document.permissionsPolicy || document.featurePolicy;
```
Comment (jpchase) [inline public/js/app.js:106]: As discussed offline, I think that `document.policy` was only briefly supported (for one release?). It seems fine to not support that here.

File: public/demos/animations.html [hunk 10]
```diff
@@ -4,7 +4,7 @@
   <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <meta http-equiv="X-UA-Compatible" content="ie=edge">
-  <title>Feature Policy: animations example</title>
+  <title>Permissions Policy: animations example</title>
```
Comment (jpchase) [inline public/demos/animations.html:7]: According to the [Chromium source](https://source.chromium.org/chromium/chromium/src/+/master:out/Debug/gen/third_party/blink/common/feature_policy/document_policy_features.cc;l=70;drc=5c28732f7b15df34c9fe70dae7f03c791c469df5), animations are in Document, not Permissions Policy.

File: public/js/policies.json [hunk 11]
```diff
@@ -101,14 +101,14 @@
   "url": "/demos/animations.html",
   "what": "Restricts the set of CSS properties which can be animated to opacity, transform, and filter.",
   "why": "Ensures smooth animations, by only allowing those properties which can be animated on the GPU using the hardware acceleration.",
-  "policyType": "Feature-Policy",
+  "policyType": "Permissions-Policy",
```
Comment (jpchase) [inline public/js/policies.json:104]: Ditto for animations being under Document Policy.

File: public/index.html [hunk 12]
```diff
@@ -4,7 +4,7 @@
   <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <meta http-equiv="X-UA-Compatible" content="ie=edge">
-  <title>Feature Policy Kitchen Sink</title>
+  <title>Permissions Policy Kitchen Sink</title>
```
Comment (jpchase) [inline public/index.html:7]: There's demo pages for both Permissions and Document Policy, so that should clearer.

**Review by jpchase — CHANGES_REQUESTED** [review]

File: README.md [hunk 13]
```diff
@@ -1,6 +1,6 @@
-## Feature Policy Demos
+## Permissions Policy Demos
```
[the remark this answers is not in the document]
Comment (huchenlei) [reply README.md:1]: Done.

File: README.md [hunk 14]
```diff
@@ -1,6 +1,6 @@
...
 
-Kitchen sink of [feature policy](https://www.chromestatus.com/feature/5694225681219584) demos.
+Kitchen sink of [permissions policy](https://www.chromestatus.com/feature/5745992911552512) demos.
```
[the remark this answers is not in the document]
Comment (huchenlei) [reply README.md:3]: Done.

File: public/demos/autoplay.html [hunk 15]
```diff
@@ -19,7 +19,7 @@ <h2 id="feature-allowed-banner">
   <span id="allowfeature">...</span>
 </h2>
 
-<p>The video below uses <code>autoplay</code> attribute. Reload the page with the feature policy on/off to see the effect on it auto playing.</p>
+<p>The video below uses <code>autoplay</code> attribute. Reload the page with the permissions policy on/off to see the effect on it auto playing.</p>
```
[the remark this answers is not in the document]
Comment (huchenlei) [reply public/demos/autoplay.html:22]: Done.

File: public/js/app.js [hunk 16]
```diff
@@ -30,15 +30,14 @@ const POLICY_TYPE_TO_LABEL = {
 
 /**
  * Dynamically loads policy demo page based off current (deep link) url.
- * @return {!Object} Feature policy info.
+ * @return {!Object} Permissions policy info.
```
[the remark this answers is not in the document]
Comment (huchenlei) [reply public/js/app.js:33]: Done.

File: server.mjs [hunk 17]
```diff
@@ -20,6 +20,8 @@ import fs from 'fs';
 import express from 'express';
 import bodyParser from 'body-parser';
 
+import {FeaturePolicyHeader, PermissionsPolicyHeader} from './permissions-policy-header.mjs';
```
[the remark this answers is not in the document]
Comment (huchenlei) [reply server.mjs:23]: Done.

File: public/js/shared.js (hunk 8 above)
Comment (huchenlei) [reply public/js/shared.js:30]: Done.

File: public/js/app.js (hunk 9 above)
Comment (huchenlei) [reply public/js/app.js:106]: Done.

File: public/demos/animations.html (hunk 10 above)
Comment (huchenlei) [reply public/demos/animations.html:7]: Done.

File: public/js/policies.json (hunk 11 above)
Comment (huchenlei) [reply public/js/policies.json:104]: Done.

File: public/index.html (hunk 12 above)
Comment (huchenlei) [reply public/index.html:7]: Done.

File: permissions-policy-header.mjs (hunk 1 above)
Comment (huchenlei) [reply permissions-policy-header.mjs:114]: Done.

File: permissions-policy-header.mjs (hunk 2 above)
Comment (huchenlei) [reply permissions-policy-header.mjs:46]: `Map.entires()` returns Iterator object, which does not have `map` method. The spread here is necessary to convert Iterator to Array.

File: public/js/app.js (hunk 5 above)
Comment (huchenlei) [reply public/js/app.js:76]: Done.

File: public/js/shared.js (hunk 6 above)
Comment (huchenlei) [reply public/js/shared.js:128]: Done.

File: permissions-policy-header.mjs (hunk 2 above)
Comment (jpchase) [reply permissions-policy-header.mjs:46]: Ah, I get it. I haven't used Map much, but a quick search suggests using the spread is most common way to handle this case.

**Review by jpchase — APPROVED** [review]
LGTM

**Pull request merged**